common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationSupport.java

```diff
@@ -46,14 +49,11 @@ public static UTF8String[] exec(final UTF8String s, final UTF8String d, final in
       }
     }
     public static String genCode(final String s, final String d, final int collationId) {
-      CollationFactory.Collation collation = CollationFactory.fetchCollation(collationId);
       String expr = "CollationSupport.StringSplitSQL.exec";
-      if (collation.isUtf8BinaryType) {
+      if (CollationFactory.UTF8_BINARY_COLLATION_ID == collationId) {
```

Comment:
```suggestion
      if (collationId == CollationFactory.UTF8_BINARY_COLLATION_ID) {
```

(or update the others - pick according to your preference, but let's keep it consistent across the file)